stl/inc/optional

```diff
@@ -164,8 +168,8 @@ public:
     using value_type = _Ty;
 
     // constructors [optional.object.ctor]
-    constexpr optional() noexcept : _Mybase{} {}
-    constexpr optional(nullopt_t) noexcept : _Mybase{} {}
+    constexpr optional() noexcept {}
+    constexpr optional(nullopt_t) noexcept {}

```

Comment:
It is possible to write a test for this non-Standard performance optimization (scribble bytes, construct an `optional`, verify that the scribbles are unchanged), although I don't think we need to add such a test. No change requested.

Reply:
I'm not fond of such "reuse storage for an object and hope that inspecting the object representation shows unchanged values" tests: nothing in the object model guarantees that they will work.

Reply:
The tests (which I wrote aeons ago) are definitely bad kitties.